docs/user-guide/builtins.md

```diff
@@ -116,6 +116,74 @@ Such named modules will always then be available under the
     Please see the documentation (linked above) about restrictions and gotchas
     when configuring how JavaScript modules are made available to PyScript.
 
+### `pyscript.fetch`
+
+A common task is to `fetch` data from the web via HTTP requests. The
+`pyscript.fetch` function provides a uniform way to achieve this in both
+Pyodide and MicroPython. It is closely modelled on the
+[Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) found
+in browsers with some important Pythonic differences.
+
+The simple use case is to pass in a URL and `await` the response. Remember, in
+order to use `await` you must have the `async` attribute in the `script` tag
+that references your code.
+
+```python title="A simple HTTP GET with pyscript.fetch"
+from pyscript import fetch
+
+
+response = await fetch("https://example.com")
+```
+
+If the `fetch` operation _returns a response that is not deemed `OK`_ (the
+definition of which
+[can be found here](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok))
+then an exception is raised.
+
+Assuming an `OK` response, the following methods are available to you to access
+the data returned from the server:
+
+* `response.arrayBuffer()` returns a Python [memoryview](https://docs.python.org/3/library/stdtypes.html#memoryview) of the response. This is equivalent to the [`arrayBuffer()` method](https://developer.mozilla.org/en-US/docs/Web/API/Response/arrayBuffer) in the browser based `fetch` API.
+* `response.blob()` returns a JavaScript [`blob`](https://developer.mozilla.org/en-US/docs/Web/API/Response/blob) version of the response. This is equivalent
+to the [`blob()` method](https://developer.mozilla.org/en-US/docs/Web/API/Response/blob) in the browser based `fetch` API.
+* `response.bytearray()` returns a Python [`bytearray`](https://docs.python.org/3/library/stdtypes.html#bytearray) version of the response.
+* `response.json()` returns a Python datastructure representing a JSON serialised payload in the response.
+* `response.text()` returns a Python string version of the response.
+
+!!! warning
+
+    Unlike the browser based `fetch` API, you **do not need to await** the
+    methods listed above, in order to get the data from the response.
```

Comment:
wrong, that's the only thing you need to await here ... those are mandatory and async by all means, please see the counter example ... we offer one await but it's going to be a one-off operation.

Not sure why it's upside down here, please let's fix this.

Reply:
Done.